pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -1,34 +1,175 @@
 """
 Copyright 2020 Google LLC
-Copyright 2020 PerfectVIPs Inc.
-
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
-http://www.apache.org/licenses/LICENSE-2.0
+     http://www.apache.org/licenses/LICENSE-2.0
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
-Regression script for RISC-V random instruction generator
 
 """
+import logging
+import riscv_core_setting
+from riscv_instr_pkg import *
+from riscv_core_setting import *
 
 
 class riscv_instr_gen_config:
+   
     def __init__(self):
-        self.enable_floating_point = 1
-        self.disable_compressed_instr = 1
-        self.num_of_test = 3
-        self.no_fence = 1
-        self.enable_sfence = 0
-        self.reserved_regs = []
-        self.enable_illegal_csr_instruction = 0
-        self.enable_access_invalid_csr_level = 0
-        self.invalid_priv_mode_csrs = []
-        self.init_privileged_mode = "MACHINE_MODE"
-        self.no_ebreak = 1
-        self.no_dret = 1
-        self.no_csr_instr = 1
+        # dicts for exception_cause_t & interrupt_cause_t Enum classes
+        self.m_mode_exception_delegation = {}
+        self.s_mode_exception_delegation = {}
+        self.m_mode_interrupt_delegation = {}
+        self.s_mode_interrupt_delegation = {}
+        self.support_supervisor_mode = 0
         self.no_wfi = 1
+        self.min_stack_len_per_program = 10 * (riscv_core_setting.XLEN/8)
+        self.max_stack_len_per_program = 16 * (riscv_core_setting.XLEN/8)
+        self.require_signature_addr = 0
+        self.signature_addr = 0xdeadbeef        
+    
+        # lists for mode_exception & mode_interrupt
+        self.mode_exp_lst = list(map(lambda a: a.name, exception_cause_t))
+        self.mode_intrpt_lst = list(map(lambda b: b.name, interrupt_cause_t))
+    
+        # list for init_privileged_mode
+        self.init_privileged_mode = list(map(lambda mode: mode.name, privileged_mode_t))
+    
+        # list of main implemented CSRs
+        self.invalid_priv_mode_csrs = [] 
+     
+        # initialsation for mode dicts
+        for i in self.mode_exp_lst:
+          self.m_mode_exception_delegation[i] = None
+          self.s_mode_exception_delegation[i] = None
+    
+        for j in self.mode_intrpt_lst:
+          self.m_mode_interrupt_delegation[j] = None
+          self.s_mode_interrupt_delegation[j] = None
+  
+  
```

Comment:
Use a single next line after completion of function.

Reply:
cleared